six/modules/python/six.sicd/tests/test_projection_polynomials.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import os
+import sys
+
+from coda.coda_types import RowColDouble, RowColSizeT, VectorString
+from coda.coda_logging import getLogger
+import pysix.six_sicd as six_sicd
+from pysix.six_sicd import SixSicdUtilities, AreaPlaneUtility
+
+
+def getSchemaPaths(pathname):
+    schemaPaths = VectorString()
+    if pathname:
+        schemaPaths.push_back(pathname)
+    elif 'SIX_SCHEMA_PATH' in os.environ:
+        schemaPaths.push_back(os.environ['SIX_SCHEMA_PATH'])
+    else:
+        schemaPaths.push_back(six_sicd.schema_path())
+    return schemaPaths
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('Read a SICD, project the image data to '
+                                     'the output plane, and write it as an SIO')
+    parser.add_argument('--schema', help='Specifiy a schema or '
+                        'directory of schemata')
+    parser.add_argument('input', help='Input SICD')
+    args = parser.parse_args()
+
+    if not os.path.isfile(args.input):
+        print('{} is not a file'.format(args.input))
+        sys.exit(1)
+
+    if args.schema and not os.path.exists(args.schema):
+        print('{} does not exist'.format(args.schema))
+        sys.exit(1)
+
+    schemaPaths = getSchemaPaths(args.schema)
+    _, complexData = six_sicd.read(args.input, schemaPaths)
+
+    polynomialFitter = SixSicdUtilities.getPolynomialFitter(complexData)
+    sampleSpacing = RowColDouble(complexData.grid.row.sampleSpacing,
+                                 complexData.grid.col.sampleSpacing)
+    offset = RowColSizeT(complexData.imageData.firstRow,
+                         complexData.imageData.firstCol)
+    scpPixel = RowColDouble(complexData.imageData.scpPixel.row,
+                            complexData.imageData.scpPixel.col)
+
+    toSlantRow, toSlantCol = polynomialFitter.fitOutputToSlantPolynomials(
+        offset, scpPixel, scpPixel, sampleSpacing, 9, 9)
```

Comment:
Make this 9 an input argument and default it to 3 (so we don't overfit).  Just one input argument to use for all four spots where there's a 9 is good.

Reply:
Done